storage/innobase/btr/btr0sea.cc

```diff
@@ -643,7 +643,7 @@ btr_search_update_hash_ref(
 				rec_get_offsets(rec, index, offsets_,
 						ULINT_UNDEFINED, &heap),
 				block->curr_n_fields,
-				block->curr_n_bytes, index->id);
+				block->curr_n_bytes, index);
```

Comment:
On a first thinking, this may be needed, so that we will be able to fetch default values.

But, on a second thinking, the instant ADD COLUMN only affects the clustered index records. 

The format of secondary index records will be unchanged: they will continue to contain the secondary key columns followed by the not-yet-mentioned PK columns.

The format of clustered index records will change. Yes, my understanding is that the adaptive hash index can be built on any B-tree index leaf page, but only up to `dict_index_get_n_unique_in_tree(index)` columns. This would be only the PRIMARY KEY columns in the clustered index, not DB_TRX_ID,DB_ROLL_PTR or any non-PK columns.

So, Jan is correct that the changes to `btr0sea.cc` are *NOT* needed and should be reverted.